pkg/apis/mongodb/v1/mongodb_types.go

```diff
@@ -56,6 +65,36 @@ type StatefulSetConfiguration struct {
 	Spec appsv1.StatefulSetSpec `json:"spec"`
 }
 
+// MongodConfiguration holds the optional mongod configuration
+// that should be merged with the operator created one.
+//
+// The CRD generator does not support map[string]interface{}
+// on the top level and hence we need to work around this with
+// a wrapping struct.
+type MongodConfiguration struct {
+	Object map[string]interface{} `json:"-"`
```

Comment:
The Operator SDK does not support `type MongodConfiguration map[string]interface{}` as we use in enterprise. According to https://github.com/kubernetes-sigs/controller-tools/pull/126 and other threads this is a deliberate decision.

They instead encourage people to use other types, for example `json.RawMessage`. That would work for us but I opted to use a wrapping struct instead to simplify the programmatic creation of MongoDB resource definitions. If we were to use `json.RawMessage`, our tests would have to marshal the JSON twice and it would be painful for anybody who wants to programmatically interface with our operator.